Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -512,6 +512,77 @@ extension NSManagedObjectContext: MessageDatabaseSession {
         message.pinnedBy = nil
         message.pinExpires = nil
     }
+
+    /// Adds the reaction for the current user to the message with id `messageId`
+    ///
+    /// Notes:
+    /// - The reaction is added to the database and it updates the message `reactionScores` property
+    /// - This method will throw if there is no current user set
+    /// - If the message is not found, there will be no side effect and the method will return `nil`
+    /// - If a reaction for the same user, type and message exists
+    func addReaction(
+        to messageId: MessageId,
+        type: MessageReactionType,
+        score: Int,
+        extraData: [String: RawJSON]
+    ) throws -> MessageReactionDTO? {
+        guard let currentUserDTO = currentUser else {
+            throw ClientError.CurrentUserDoesNotExist()
+        }
+
+        guard let message = MessageDTO.load(id: messageId, context: self) else {
+            return nil
+        }
```

Comment:
`throws -> MessageReactionDTO?` is a bit redundant, it should be `throws -> MessageReactionDTO` and in this case, it should return an Error as well, something like `ClientError.MessageDoesNotExit()`

Reply:
debatable point, I think its better to rely on the API call to settle the score here

Reply:
This is not an API Call tho, it is a DB Write, it just makes the code harder to handle. Not only do you need to catch the error, but also handle a `nil`. This is not very "swifty"